Code/Mantid/Framework/DataObjects/src/ReflectometryTransform.cpp

```diff
@@ -223,8 +229,27 @@ Mantid::API::MatrixWorkspace_sptr ReflectometryTransform::execute(
   return ws;
 }
 
+boost::optional<boost::shared_ptr<Mantid::DataObjects::TableWorkspace>>
+createTable(boost::optional<
```

Comment:
This call should happen closer to where the Table is first needed. Put it just before the outer for loop (the loop over the number of histograms)